src/task/task.cpp

```diff
@@ -1,51 +1,142 @@
 #include "task/task.h"
-#include "task/task_manager.h"
+
+#include <utility>
 
 #include "optimizer/cost_model/trivial_cost_model.h"
+#include "parser/expression/constant_value_expression.h"
+#include "task/task_manager.h"
 #include "util/query_exec_util.h"
 
 namespace noisepage::task {
 
+Task::Task(const catalog::db_oid_t db_oid, std::string &&query_text, const transaction::TransactionPolicy policy,
+           const common::ManagedPointer<common::FutureDummy> sync)
+    : db_oid_(db_oid), query_text_(std::move(query_text)), policy_(policy), sync_(sync) {}
+
+std::unique_ptr<TaskDDL> TaskDDL::Builder::Build() {
+  NOISEPAGE_ASSERT(db_oid_.has_value(), "Database OID must be specified.");
+  NOISEPAGE_ASSERT(*db_oid_ != catalog::INVALID_DATABASE_OID, "You're doing something hacky. Get a real database OID.");
+  NOISEPAGE_ASSERT(query_text_.has_value(), "Query text must be specified.");
+  NOISEPAGE_ASSERT(policy_.has_value(), "Transaction policy must be specified.");
+  return std::unique_ptr<TaskDDL>(new TaskDDL(*db_oid_, std::move(*query_text_), *policy_, sync_));
```

Comment:
`std::make_unique()`?

Reply:
This would require making the ctor public, are you in favor of that approach?
```
/usr/include/c++/9/bits/unique_ptr.h:857:30: error: ‘noisepage::task::TaskDDL::TaskDDL(noisepage::catalog::db_oid_t, std::string&&, noisepage::transaction::TransactionPolicy, noisepage::common::ManagedPointer<noisepage::common::Future<noisepage::common::FutureDummyResult> >)’ is private within this context
```

Reply:
Whoops, I need to pay closer attention to the context, that is the whole point of the builder... my bad! No I like the current approach. This is resolved.